lib/neo4j/shared/declared_property.rb

```diff
@@ -9,12 +11,46 @@ class IllegalPropertyError < StandardError; end
 
     def initialize(name, options = {})
       fail IllegalPropertyError, "#{name} is an illegal property" if ILLEGAL_PROPS.include?(name.to_s)
-      @name = @name_sym = name
+      fail TypeError, "can't convert #{name.class} into Symbol" unless name.respond_to?(:to_sym)
+      @name = @name_sym = name.to_sym
       @name_string = name.to_s
       @options = options
       fail_invalid_options!
     end
 
+    # Compare attribute definitions
+    #
+    # @example
+    #   attribute_definition <=> other
+    #
+    # @param [Neo4j::Shared::AttributeDefinition, Object] other The other
+    #   attribute definition to compare with.
+    #
+    # @return [-1, 0, 1, nil]
+    def <=>(other)
+      return nil unless other.instance_of? self.class
+      return nil if name == other.name && options != other.options
+      self.name.to_s <=> other.name.to_s
```

Comment:
Maybe this should be `self.to_s <=> other.to_s` since there is a `to_s` on line 42?